### ISSUE TITLE
RFE: Possibility for grading quality of TLS connection

### ISSUE THREAD

**Issue by hardfalcon** [april/certainly-something#9]
[SSLeuth](https://addons.mozilla.org/firefox/addon/ssleuth/) offered a simple grading of the TLS connections quality, based upon factors like "strength of the symmetric encryption", "strength of the key exchange", "strength of the MAC/AEAD", "forward secrecy", "extended validation", "certificate status", etc, where the wheight for every single factor could be [configured](https://addons.cdn.mozilla.net/user-media/previews/full/135/135336.png).

A similar feature would be nice to have, perhaps even rendering the grading directly onto the extension's icon in the URL bar.

**Comment by april** [comment]
This is a pretty big undertaking, and might be better as a separate project.

I would say about 80% or so of what SSLeuth looks at is possible with the new API. Is it still relevant when services like [SSL Labs](https://www.ssllabs.com/ssltest/) will provide a much more in-depth analysis?

**Comment by rugk** [comment]
My main use case for SSLeuth was to quickly see if a connection was not the best. I just cannot and do not scan each site I visit with SSLLabs.
So, SSLeuth shows you:
* info for each site I visit, directly in the URL bar, i.e. it's much faster
* the garding of the *current connection*, not a general assesment
* you may adjust your behavior/not visit the site, if the grading is too bad

**Comment by hardfalcon** [comment]
@april: Exactly what @rugk said/wrote.

**Comment by km-js** [comment]
hey @april I would like to contribute on this one being an outreachy aspirant. Can you please suggest me which code to touch?

**Comment by april** [comment]
I don't think this would be a particularly good task for an outreachy aspirant, especially given that I don't even know how I would grade connections at this point.

**Comment by noahwalugembe** [comment]
Hi @april .
I believe i can work on this can i go on.

**Comment by april** [comment]
I don't even know how I would do this. If you want to write up a proposal I could certainly take a look at it, but the scope of this is probably far beyond any other bug.

**Comment by noahwalugembe** [comment]
@april Oky April i am going to make a write up I just finished setting up. This commands{
$ npm run-script watch
$ web-ext run --browser-console -s build --start-url 'https://badssl.com/'} 
are working  but i don't Know how to access the graphical user interface for certainly-something 
some thing like this blow{
https://camo.githubusercontent.com/1b437f9f8565cca863b268f2720163c09c997173/68747470733a2f2f692e696d6775722e636f6d2f474562763433342e706e67

}

**Comment by noahwalugembe** [comment]
> I don't even know how I would do this. If you want to write up a proposal I could certainly take a look at it, but the scope of this is probably far beyond any other bug.

I have created the proposal for grading quality of TLS connection. Pleas allow me to work on it.

[Proposal.pdf](https://github.com/april/certainly-something/files/2951280/Proposal.pdf)

**Comment by noahwalugembe** [comment]
> I don't even know how I would do this. If you want to write up a proposal I could certainly take a look at it, but the scope of this is probably far beyond any other bug.

@april I have created the proposal for grading quality of TLS connection. Pleas allow me to work on it.

[Proposal.pdf](https://github.com/april/certainly-something/files/2951280/Proposal.pdf)

**Comment by noahwalugembe** [comment]
> I don't even know how I would do this. If you want to write up a proposal I could certainly take a look at it, but the scope of this is probably far beyond any other bug.

@april I have created the proposal for grading quality of TLS connection. Pleas allow me to work on it.

[Proposal.pdf](https://github.com/april/certainly-something/files/2951280/Proposal.pdf)

**Comment by rugk** [comment]
So you want to use HTTP Observatory. The things I see here:
* privacy impact: the user has to know their domain is sent to a third-party service
* more analysis: HTTP Observatory analyses more than just HTTPS. (it's a big part, but it also takes HTTP headers etc. into account) So this can be a little misleading...

And personally, I still consider this out of scope of this add-on. It would possibly better fit into a new add-on...

**Comment by noahwalugembe** [comment]
> So you want to use HTTP Observatory. The things I see here:
> 
> * privacy impact: the user has to know their domain is sent to a third-party service
> * more analysis: HTTP Observatory analyses more than just HTTPS. (it's a big part, but it also takes HTTP headers etc. into account) So this can be a little misleading...
> 
> And personally, I still consider this out of scope of this add-on. It would possibly better fit into a new add-on...

Hi @april  I believe we can prompt a user reject or allow us send his or her domain name to a third-party service.
So far i have most of the code working and and i can start making commits it wont cost me much time to accomplish. 
Pleas allow me  do this task.

**Comment by rugk** [comment]
I am not @april I am just watching this repo as I like the add-on giving my two cents. So let's first see what @april actually says...

**Comment by april** [comment]
The HTTP Observatory doesn't do any grading of TLS, nor does the TLS Observatory. Nothing in this proposal would address the request in the issue, @noahwalugembe.

Sorry, I don't think this issue is at an appropriate complexity level for you to address.

**Comment by noahwalugembe** [comment]
> The HTTP Observatory doesn't do any grading of TLS, nor does the TLS Observatory. Nothing in this proposal would address the request in the issue, @noahwalugembe. Further, tools like SSL Labs and the TLS Observatory also only address available cipher suites and protocols, but they don't grade what the browser is actually using.
> 
> Sorry, I don't think this issue is at an appropriate complexity level for you to address.

Thank april for your advise. Is it okay if i can ask you to get form me some new bugs which i can work on this week. Pleas give me a hand. I really need to contribute so as to qualify for internship with outreachy.
I will my pleasure to  work with you on this project.

**Comment by FranklinYu** [comment]
The grading is both subjective and transient (what is secure now may not be secure in 2 years). I don’t think this function is in scope of the project.
